pkg/internal/apply/layerApplier.go

```diff
@@ -154,6 +155,51 @@ func (a KubectlLayerApplier) decodeAddons(layer layers.Layer,
 	}
 }
 
+func (a KubectlLayerApplier) addOwnerRefs(layer layers.Layer, hrs []*helmopv1.HelmRelease) error {
+	for i, hr := range hrs {
+		a.logDebug("Adding owner ref to HelmRelease for AddonsLayer", layer, "index", i, "helmRelease", hr)
+		err := controllerutil.SetControllerReference(layer.GetAddonsLayer(), hr, a.scheme)
+		if err != nil {
+			// could not apply owner ref for object
+			return fmt.Errorf("Unable to apply owner reference for AddonsLayer '%s' to HelmRelease '%s'", layer.GetName(), getLabel(hr))
+		}
+	}
+	return nil
+}
+
+func (a KubectlLayerApplier) getHelmRelease(layer layers.Layer, hr *helmopv1.HelmRelease) (*helmopv1.HelmRelease, error) {
```

Comment:
I think i'm missing something here. You pass in a helmrelease and then get the helmrelease. Is this to ensure you have the latest?

Reply:
I'm passing in the HelmRelease created from parsing YAML in the sourceDirectory and getting a matching HelmRelease queried from the KubeAPI - Kraan will compare them against each other to determine if it needs to apply source YAML changes to this HelmRelease.